src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs

```diff
@@ -172,22 +172,14 @@ private PSVirtualNetworkGatewayConnection CreateVirtualNetworkGatewayConnection(
             vnetGatewayConnection.ConnectionType = this.ConnectionType;
             vnetGatewayConnection.RoutingWeight = this.RoutingWeight;
             vnetGatewayConnection.SharedKey = this.SharedKey;
-            
-            if (!string.IsNullOrEmpty(this.EnableBgp))
-            {
-                vnetGatewayConnection.EnableBgp = bool.Parse(this.EnableBgp);
-            }
-            else
-            {
-                vnetGatewayConnection.EnableBgp = false;
-            }
+
+            vnetGatewayConnection.EnableBgp = this.EnableBgp;
```

Comment:
@markcowl  Is this MyInvocation.BoundParameters a dictionary of provided parameters? (key = parametername, value = parametervalue)? I was not aware of this capability - this might change my entire design